### PR TITLE
feat(stock) searching inventory by a barcode scanner

### LINKED ISSUE #3396: Search by barcode on Stock Registries
The stock registries could use a search by barcode feature, similar to our invoice, cash payments, and patient registry.  This feature would open a scan barcode modal and the user would scan the barcode of the lot in question.  This would then filter the registry looking for the lot.  See below:

featurerequestsearchbybarcodeinstockregistry
_Fig 1: Add the Barcode Search to the Stock Registry_

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #3396

### CODE REVIEW

File: client/src/js/components/bhBarcodeScanner.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ angular.module('bhima.components')
     controller  : bhBarcodeScanner,
     bindings    : {
       onScanCallback : '&',
+      shouldSearch : '<',
```
Comment (mbayopanda) [inline client/src/js/components/bhBarcodeScanner.js:7]: does the `should-search` attribute is required ? if not, could you use 
```
shouldSearch : '<?',
```
for notifying that it is optional

File: client/src/modules/stock/inventories/registry.html [hunk 2]
```diff
@@ -48,6 +48,13 @@
                 <span class="fa fa-file-excel-o"></span> <span translate>DOWNLOADS.EXCEL</span>
               </a>
             </li>
+
+            <li role="separator" class="divider"></li>
+            <li role="menuitem" bh-require-enterprise-setting="barcodes">
+              <a href ng-click="StockCtrl.openBarcodeScanner()" data-action="scan-barcode">
+                <span class="fa fa-barcode"></span> <span translate>BARCODE.SCAN</span>
+              </a>
+            </li>
```
Comment (mbayopanda) [inline client/src/modules/stock/inventories/registry.html:57]: 👍 excellent

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

bors r+

**Comment by jniles** [comment]
@jeremielodi, as soon as conflicts are resolved, we can merge this.

**Comment by jeremielodi** [comment]
bors r*

**Pull request merged**